libs/api/ebpf_api.cpp

```diff
@@ -2958,8 +2963,9 @@ _ebpf_object_reuse_map(_Inout_ ebpf_map_t* map) NO_EXCEPT_TRY
 
     ebpf_assert(map);
 
-    // Check if a map is already present with this pin path.
-    fd_t map_fd = ebpf_object_get(map->pin_path);
+    // If there is no map at this pin path, then we can (re)use the map.
+    fd_t map_fd = ebpf_fd_invalid;
+    result = ebpf_object_get(map->pin_path, &map_fd);
     if (map_fd == ebpf_fd_invalid) {
```

Comment:
Seems like this should check for a specific result value that means there's no map at this pin path.

Reply:
Agree, the previous code can b refactored.